pkgs/applications/misc/pagefind/default.nix

```diff
@@ -0,0 +1,98 @@
+{ lib
+, callPackage
+, rustPlatform
+, fetchFromGitHub
+, fetchNpmDeps
+, npmHooks
+, binaryen
+, gzip
+, nodejs
+, wasm-bindgen-cli
+, wasm-pack
+}:
+
+let
+
+  # https://github.com/NixOS/nixpkgs/issues/89426
+  # we directly re-use the existing fix to avoid adding
+  # more rustc versions to the cache should the fixes diverge
+  rustc-wasm = callPackage ./../../../servers/ldap/lldap/rustc-wasm.nix { };
+
+  wasm-bindgen-84 = wasm-bindgen-cli.override {
+    version = "0.2.84";
+    hash = "sha256-0rK+Yx4/Jy44Fw5VwJ3tG243ZsyOIBBehYU54XP/JGk=";
+    cargoHash = "sha256-vcpxcRlW1OKoD64owFF6mkxSqmNrvY+y3Ckn5UwEQ50=";
+  };
+
+in
+
+rustPlatform.buildRustPackage rec {
+  pname = "pagefind";
+  version = "unstable-2023-06-29"; # latest release is missing a lock file
+
+  src = fetchFromGitHub {
+    owner = "cloudcannon";
+    repo = "pagefind";
+    rev = "6de1deeeea03d15aaaad5ba318155e79b7b9db29";
+    hash = "sha256-qaf9EDqckkHR0wvFSuc7iGaN3aBZO0eNHwTX8Zna5E4=";
+  };
+
+  cargoHash = "sha256-WElWA6Q/TFue12c7jt6v/d8ddu4aJocDG8lbdhwABOc=";
```

Comment:
I think we prefer using `cargoLock.lockFile` now, but that information might be outdated

Reply:
As I understand that was a treewide workaround for git dependencies suddenly becoming unreproducible.